### PR TITLE
fix(dashboards): align health metrics overview with design spec

### PR DESCRIPTION
**Pull request by fayazg**
## Summary

- Add a rail/sidebar (Foundation Revenue, Foundation summary, Data sources) alongside the tile strip and findings list, matching the design's two-column layout
- Add a visual-only period selector (2023/2024/2025/YTD) and Export button to the page header — non-functional until a real backend supports filtering/export
- Restructure the tile strip and findings list into shared bordered cards with dividers, replacing individual per-item borders/shadows
- Fix typography, spacing, and chart visuals (dots/bar/band/tags) across tiles and finding rows to match the design's exact values
- Color findings-group headers per classification tone with a trailing divider line; thin the findings' left accent from 4px to 2px, and drop it entirely from tiles
- Swap the Engagement/Non-Members icons to match the design's assignment
- Fix a sticky-header height measurement bug (border-box vs. content-box) and related accessibility gaps (aria-disabled vs. disabled, aria-labelledby wiring on chart captions, ARIA roles on visuals)
- Extend unit test coverage across the touched components and shared utils

Refs LFXV2-3365

### CODE REVIEW

File: packages/shared/src/utils/health-metrics-overview.utils.ts [hunk 1]
```diff
@@ -70,15 +74,42 @@ export function buildHealthMetricsOverviewTiles(areaStates: HealthMetricsAreaSta
...
+const UNKNOWN_REVENUE_STREAM_META = { label: 'Other', dotClass: 'bg-gray-400' } as const;
+
+/** Builds the rail's "Foundation Revenue" legend rows — percent share and formatted total per stream. */
+export function buildHealthMetricsOverviewRevenueStreams(revenue: HealthMetricsOverviewRevenue): HealthMetricsOverviewRevenueStreamViewModel[] {
+  return revenue.streams.map((stream) => {
+    const meta = HEALTH_METRICS_OVERVIEW_REVENUE_STREAMS[stream.key] ?? UNKNOWN_REVENUE_STREAM_META;
+    return {
+      label: meta.label,
+      dotClass: meta.dotClass,
+      percent: revenue.total > 0 ? Math.round((stream.value / revenue.total) * 100) : 0,
```
[the remark this answers is not in the document]
Comment (fayazg) [reply packages/shared/src/utils/health-metrics-overview.utils.ts]: Done — `buildHealthMetricsOverviewRevenueStreams` now returns both a rounded `percent` (for the "N%" legend text) and a separate unrounded `widthPercent` (for the segmented bar's `[style.width.%]` binding), so independently-rounded segments can no longer sum to less than 100% and leave a visible gap.

See commit 4615a7a63: `health-metrics-overview-rail.component.html`'s bar segment now binds to `stream.widthPercent` instead of `stream.percent`. Added a regression test in `health-metrics-overview.utils.spec.ts` covering the exact three-equal-streams case (33/33/33 rounds display-only, `widthPercent` sums to 100).

File: apps/lfx-one/src/app/modules/dashboards/health-metrics-overview/health-metrics-overview.component.html [hunk 2]
```diff
@@ -1,36 +1,101 @@
...
-    @if (hasFindings()) {
-      @for (findingGroup of findingGroups(); track findingGroup.group) {
-        <div class="flex flex-col gap-3" [attr.data-testid]="'health-metrics-overview-findings-group-' + findingGroup.group">
-          <h2 class="text-sm font-semibold text-gray-700">{{ findingGroup.group }}</h2>
-          <div class="flex flex-col gap-2">
-            @for (finding of findingGroup.findings; track finding.sortRank) {
-              <lfx-health-metrics-overview-finding-item [finding]="finding" [attr.data-testid]="'health-metrics-overview-finding-' + finding.sortRank" />
-            }
-          </div>
+  <div class="flex flex-col gap-6 lg:grid lg:grid-cols-[minmax(0,1fr)_320px] lg:items-stretch">
```
[the remark this answers is not in the document]
Comment (fayazg) [reply apps/lfx-one/src/app/modules/dashboards/health-metrics-overview/health-metrics-overview.component.html:65]: Thanks for flagging this — I can see why the two-column grid starting after the tile strip looks like the rail should span both blocks instead.

This is intentional and matches the design mockup's actual rendered structure, not an accidental escape from the grid: the design's default `renderB()` view also renders the tile strip as its own full-width block above a separate two-column region (main content + rail), rather than putting the tile strip inside that region's grid. The rail is only meant to run alongside the findings list, not alongside the tile strip above it — the current markup (tile-strip `<section>` full width, then a `lg:grid-cols-[minmax(0,1fr)_320px]` wrapper containing just the findings list and the rail) reproduces that same two-tier layout.

No code change here. Happy to revisit if you have a different read on the mockup.

**Comment by fayazg** [comment]
## Review Feedback Addressed

Commit: 4615a7a639308bbfb7a58409de0a87cf1ba83cf5

Copilot's review posted 2 threaded comments and 8 additional suppressed findings in the review body. All 10 are addressed below.

### Changes Made
- **`packages/shared/src/utils/health-metrics-overview.utils.ts`** (threaded): `buildHealthMetricsOverviewRevenueStreams` now returns an unrounded `widthPercent` for the bar segment alongside the rounded `percent` used for the legend text, so independently-rounded segments can no longer sum short of 100%.
- **`.../health-metrics-overview-finding-item.component.ts`** (suppressed): dot groups now get a proportional budget of `MAX_RENDERED_DOTS` via a largest-remainder allocation (`allocateDotsBudget`), instead of each group independently claiming up to the full cap before the flattened row was truncated — which let one large/full group crowd out later groups entirely.
- **`health-metrics-overview.component.html`** (suppressed): the header controls wrapper is now `flex flex-wrap` so the project selector, period buttons, and Export button wrap instead of overflowing at phone width.
- **`health-metrics-overview.component.ts` + `.spec.ts`** (suppressed): added a test exercising the `ResizeObserver` path directly (stubs the global, invokes the observer callback with a `borderBoxSize` entry, asserts the rail's computed `top` offset reflects it).
- **`apps/lfx-one/tailwind.config.js`** (suppressed): safelisted `HEALTH_METRICS_OVERVIEW_REVENUE_STREAMS`'s `dotClass` values plus the `UNKNOWN_REVENUE_STREAM_META` fallback's `bg-gray-400`, since they're assembled at runtime from a `@lfx-one/shared` constant outside this config's `content` glob.
- **`health-metrics-overview.utils.ts`** (suppressed, 3 findings): guarded all three classification/revenue-stream lookups (`resolvedClassification`, `resolveHealthMetricsOverviewGroupMeta`, `buildHealthMetricsOverviewRevenueStreams`'s stream-key fallback) with `Object.hasOwn` instead of truthy/nullish-coalescing checks, which incorrectly treated inherited prototype keys (e.g. `toString`, `constructor`) as present.

### No Change Needed
- **`health-metrics-overview.component.html:65`** (threaded — tile-strip/rail grid placement): the tile strip renders as its own full-width block above a separate two-column region (findings + rail), matching the design mockup's actual `renderB()` structure. The rail is only meant to run alongside the findings list, not the tile strip above it — current markup is intentional, not an accidental grid escape.
- **`health-metrics-overview.component.html:38`** (suppressed — `aria-disabled` on the not-yet-wired project/period/export controls): deliberate trade-off. These controls are visual-only placeholders (no backend to filter by yet); `aria-disabled` (not the native `disabled` attribute) keeps them focusable and in the accessibility tree so `aria-pressed` still announces the selected period to assistive tech, at the cost of remaining click/keyboard-activatable as a no-op. Will revisit once the controls are wired to real behavior.

### Threads Resolved
2 of 2 GitHub-threaded comments resolved. The 8 suppressed findings had no separate thread to resolve — addressed above instead.

File: apps/lfx-one/src/app/modules/dashboards/health-metrics-overview/health-metrics-overview-rail/health-metrics-overview-rail.component.html [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  <div class="flex flex-col p-[20px]">
+    <div class="mb-[16px] flex items-center gap-[10px]">
+      <span class="text-[12px] font-semibold uppercase text-gray-600">Foundation Revenue</span>
+      <span class="h-px flex-1 bg-gray-300" aria-hidden="true"></span>
+    </div>
+    <span class="text-[20px] font-semibold text-gray-900">{{ totalLabel() }}</span>
+
+    <div class="my-4 flex h-[6px] overflow-hidden rounded-full bg-gray-100" role="img" aria-label="Revenue by stream">
+      @for (stream of streams(); track stream.label) {
+        <div [ngClass]="stream.dotClass" [style.width.%]="stream.widthPercent" aria-hidden="true"></div>
```
[the remark this answers is not in the document]
Comment (fayazg) [reply apps/lfx-one/src/app/modules/dashboards/health-metrics-overview/health-metrics-overview-rail/health-metrics-overview-rail.component.html]: Done — preserved the raw stream key in the view model and both `@for` loops now track by it.

`buildHealthMetricsOverviewRevenueStreams` now includes `key: stream.key` on `HealthMetricsOverviewRevenueStreamViewModel`, and `health-metrics-overview-rail.component.html`'s bar-segment and legend loops now use `track stream.key` instead of `track stream.label`. Added a regression test asserting two streams that both degrade to the fallback `'Other'` label still keep their own distinct keys.

See commit 8d84568d0.

**Comment by fayazg** [comment]
## Review Feedback Addressed (iteration 2)

Commit: 8d84568d05490e11acb51ca4a6a3d97dcaefefec

### Changes Made
- **health-metrics-overview.interface.ts**: added a `key` field to `HealthMetricsOverviewRevenueStreamViewModel`, holding the raw `stream.key` (per copilot-pull-request-reviewer[bot])
- **health-metrics-overview.utils.ts**: `buildHealthMetricsOverviewRevenueStreams` now populates `key` from `stream.key`
- **health-metrics-overview-rail.component.html**: both `@for` loops (bar segments and legend rows) now `track stream.key` instead of `track stream.label` — two degraded/unknown streams previously shared the fallback `'Other'` label, which Angular could report as duplicate tracking keys
- **health-metrics-overview.utils.spec.ts**: added a regression test asserting distinct keys survive even when two streams degrade to the same fallback label

### Threads Resolved
1 of 1 unresolved thread addressed in this iteration.

**Comment by dealako** [comment]
@fayazg thanks for the PR — this brings the Health Metrics Overview dashboard in line with the design spec: refreshed layout (sticky header, tile strip, findings list, revenue rail), accessibility work (aria-pressed, aria-disabled, scoped ids), and the supporting view-model utilities in `@lfx-one/shared`. The scope is well contained to the dashboards module and its shared utils, and the commit history shows the two Copilot review rounds were already addressed (rounding gap between segmented bar width and legend percent fixed via a separate `widthPercent`; duplicate `@for` tracking key for degraded revenue streams fixed by tracking on the raw `stream.key`).

I ran a full three-dimension review (security/privacy, correctness/performance/tests, style/docs) across the whole diff. Nothing blocking came out of it.

🔴 Blocking: 0 issues
🟡 Minor: 0 issues
⚪ Nit: 1 issue: missing test coverage for a bar visual whose `parts` array is empty
❔ Question: 0 items

### Nit

**Untested edge case: bar with an empty `parts` array** — `health-metrics-overview-finding-item.component.ts` (`toBarViewModel`, ~L111–123): when `bar.parts` has no matching-tone parts, the code falls back to summing all of `bar.parts`; if that's also empty, `fillPercent` correctly resolves to `0` via the `reduce` initial value, so there's no crash. But no test exercises this path — worth a small regression test asserting a 0% bar renders instead of relying on inference, since `HealthMetricsFindingVisualBar.parts` isn't typed as non-empty.

### Reconciliation with Copilot's review

Copilot's grid-layout comment (rail sitting below the tile strip rather than beside it) was already addressed in-thread as intentional — matches the design's actual rendered structure per your reply. Agreed with that call after checking the layout structure directly. Copilot's rounding and duplicate-tracking-key findings were both fixed in later commits (`4615a7a`, `8d84568d0`) — confirmed those fixes are in place and correct.

I also spot-checked a style-review finding about `[class]` vs `[ngClass]` on the group icon in `health-metrics-overview.component.html` — dropped it after checking: the sibling tile component (`health-metrics-overview-tile.component.html`) uses the same `[class]="...icon"` pattern for icon strings, so it's the established convention in this module, not an inconsistency.

No security, privacy, or data-residency findings. No merge conflicts.

✅ **Approved**

**Review by dealako — APPROVED** [review]
Approving per the consolidated dd-pr-review pass (see summary comment). No blocking, security, or privacy findings; one nit (untested empty-`parts` edge case in the bar view model) noted in the summary. No merge conflicts.
